Automattic/jetpack
# Social: Add feature flag management

This updates the feature flag check to use the plan feature added in D155553-code

Fixes https://github.com/Automattic/jetpack-reach/issues/455

## Proposed changes:

* This change adds a feature flag for `useEditorPreview`
* Also added helpers so next time we just have to update an array in `projects/packages/publicize/src/jetpack-social-settings/class-settings.php` to add new feature flags.

### Other information:


## Jetpack product discussion


https://github.com/Automattic/jetpack/blob/e79e535674be5d16af74f01f445cfd48bb003982/projects/packages/publicize/src/jetpack-social-settings/class-settings.php
## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Go to editor. You should not see the `New modal trigger goes here` span below the connection management link.
* Run `jetpack docker wp option update jetpack_social_use_editor_preview true` and refresh the editor.
* You should see the `New modal trigger goes here` span below the connection management link.
* Make sure you test this with Jetpack and Social only too.
